packages/cojson/src/sync.ts

```diff
@@ -474,31 +422,37 @@ export class SyncManager {
       }
     }
 
-    await this.tellUntoldKnownStateIncludingDependencies(msg.id, peer);
-    await this.sendNewContentIncludingDependencies(msg.id, peer);
+    if (entry.state.type === "available") {
+      await this.tellUntoldKnownStateIncludingDependencies(msg.id, peer);
+      await this.sendNewContentIncludingDependencies(msg.id, peer);
+    }
   }
 
   async handleKnownState(msg: KnownStateMessage, peer: PeerState) {
-    let entry = this.local.coValues[msg.id];
+    const entry = this.local.coValuesStore.get(msg.id);
 
     peer.dispatchToKnownStates({
       type: "COMBINE_WITH",
       id: msg.id,
       value: knownStateIn(msg),
     });
 
-    if (!entry) {
+    if (entry.state.type === "unknown" || entry.state.type === "unavailable") {
       if (msg.asDependencyOf) {
-        if (this.local.coValues[msg.asDependencyOf]) {
-          this.local
-            .loadCoValueCore(msg.id, { dontLoadFrom: peer.id })
-            .catch((e) => {
-              console.error(
-                `Error loading coValue ${msg.id} to create loading state, as dependency of ${msg.asDependencyOf}`,
-                e,
-              );
-            });
-          entry = this.local.coValues[msg.id]!; // must exist after loadCoValueCore
+        const dependencyEntry = this.local.coValuesStore.get(
+          msg.asDependencyOf,
+        );
+
+        if (
+          dependencyEntry.state.type === "available" ||
+          dependencyEntry.state.type === "loading"
+        ) {
+          this.local.loadCoValueCore(msg.id, peer.id).catch((e) => {
+            console.error(
+              `Error loading coValue ${msg.id} to create loading state, as dependency of ${msg.asDependencyOf}`,
+              e,
+            );
+          });
```

Comment:
@aeplay 

Not sure that the new state checks I've added are reflecting the previous intention here.